### ISSUE TITLE
GT06 undocumented(?) Information Packet 0x21: Door alarm

### ISSUE THREAD

**Issue by CyberPK** [traccar/traccar#4511]
Hello,
I do not own the official procotol documentation for my device but found these on internet: [https://filebin.net/ufd7a1i2ifum7ylj]
Apparently it uses the gt06 protocol, and adhere to all the specification working with traccar.
I was missing the door alarm notification, so I looked at the raw data and found this packeks:

1) door alarm ON: `7878082114040b0e3b3a0103bac3670d0a`
2) door alarm OFF: `7878082114040b0f00030003bbaab50d0a`

So, because it is undocumented and looking at the code, it is not managed, I've tried to decode it:

Packet 1)
```
7878   Start Bit
08   Packet Length
21   Information Packet 0x21
14040b0e3b3a date/time  14 year (20) | 04 month | 0b day (11) | 0e hour (14) | 3b minutes (59) | 3a seconds (59) -> 14:59:58 April 11 2020
**01       Door Alarm ON**
03ba   information SN
c367   Error Check
0d0a   Stop Bit
```

Packet 2)
```
7878   Start Bit
08   Packet Length
21   Information Packet 0x21
14040b0f0003 date/time  14 year (20) | 04 month | 0b day (11) | 0f hour (15) | 00 minutes (00) | 03 seconds (03) -> 15:00:03 April 11 2020
**00       Door Alarm OFF**
03bb  information SN
aab5   Error Check
0d0a   Stop Bit
```
I don't know if it is only limited to my device, so I don't know if you want to add this function or not.
The 0x13 packets with extend alarm, does not report the additional codes includint the door alarm.
Eventually, can you give me some hint to help me implement this simple function?

**Comment by CyberPK** [comment]
@tananaev 
I've build and tested the patch to decode this packet.
Also added a Door Open/close event and notification, and fixed some other bug on the status retaining of ignition (applied also to the door event) preventing the event to be fired.
Because the ignition event relieves on the previous status, because it is not present in every position I think to load the previous status and pass it to the new position.
Do you think it is the right approach? This way I can track when the door is closed and opened.

In other Information Packet with extended information (not applicable to my device) can be fired only an alarm which says that the door is open, but I don't know if it inform also when it closes.

**Comment by tananaev** [comment]
Create a pull request and we can discuss it there, but I would recommend to split it into separate pull requests.

**Comment by CyberPK** [comment]
this patch closes #4509 #4510 #4511
[patch.diff.txt](https://github.com/traccar/traccar/files/4470138/patch.diff.txt)

**Comment by tananaev** [comment]
Please send a pull request. We don't accept patch files.
